### ISSUE TITLE
Write-RsFolderContent for Mobile Reports (.rsmobile)

### ISSUE THREAD

**Issue by TobiasEl** [microsoft/ReportingServicesTools#330]
Hi
This module for powershell suport mobile reports (.rsmobile)?
I tried to upload mobile reports (.rsmobile) to my SSRS server and it appears a error message saying: "only reports, datasets and datasources". Wich make me think that this module only works for .rdl reports. Could you please help me? I need to upload all mobile reports (.rsmobile), datasets and datasources to my SSRS server automatically (not one by one).
Thanks.

**Comment by jtarquino** [comment]
No it only supports paginated reports (is mentioned in the documentation of the cmdlet)

**Comment by TobiasEl** [comment]
Yes. I don't know how I can help. But It would be very helpfull if ReportingServicesTools could work with .rsmobile (it works for download, but not for upload). We need it in our work team. How long do you think support for mobile reports will take? 
Thank you.

**Comment by TobiasEl** [comment]
By the way, I can't upload a .rsmobile file, but neither can I upload the datasets and datasources in the rest of the folders. It's like, when it detects that there are .rsmobile files (even in other folders), it doesn't upload anything (not even datsets and datasources).

**Comment by mashbrno** [comment]
Here is my workaround using C#, converting to PowerShell should be easy. Note that my SSRS 2019 is configured with Basic authentication.

``` c#
HttpClient client = new()
{
    BaseAddress = new Uri("https://example.com/reports/api/v2.0/catalogitems")
};
client.DefaultRequestHeaders.Authorization =
    new AuthenticationHeaderValue("Basic", "username:password");

StringBuilder sb = new("""{"ContentType":"",""");
sb.AppendFormat(@"""Name"": ""{0}"", ""Path"":""{1}"",""Content"":""", "Přehled", "/XANTIS/Přehled");
sb.Append(Convert.ToBase64String(File.ReadAllBytes(@"report.rsmobile")));
sb.Append(@""",""@odata.type"":""#Model.MobileReport""}");

StringContent payload = new(sb.ToString(), Encoding.UTF8, MediaTypeHeaderValue.Parse("application/json"));
await client.PostAsync("", payload);
```
